src/core/shape_inference/include/sparse_fill_empty_rows_shape_inference.hpp

```diff
@@ -61,10 +61,10 @@ std::vector<TRShape> shape_infer(const SparseFillEmptyRows* op,
     empty_row_indicator_shape.resize(1);
     output_indices_shape[1] = 2;  // Only 2D cases are supported
 
-    if (auto dense_shape_value = get_input_const_data_as_shape<TRShape>(op, 1, tensor_accessor)) {
+    auto dense_shape_value = get_input_const_data_as_shape<TRShape>(op, 1, tensor_accessor);
+    if (dense_shape_value && (*dense_shape_value).is_static()) {
```

Comment:
```suggestion    
    if (auto dense_shape_value = get_input_const_data_as_shape<TRShape>(op, 1, tensor_accessor); dense_shape_value && (*dense_shape_value).is_static()) {
```
Will have `if` block scope

Reply:
Done.